woocommerce-pdf-invoices-packingslips.php

```diff
@@ -467,6 +475,73 @@ public function rtl_detected() {
 			}
 		}
 	}
+	
+	/**
+	 * Get an array of all active plugins, including multisite
+	 * @return array active plugin paths
+	 */
+	public function get_active_plugins() {
+		$active_plugins = (array) apply_filters( 'active_plugins', get_option( 'active_plugins' ) );
+		if ( is_multisite() ) {
+			// get_site_option( 'active_sitewide_plugins', array() ) returns a 'reversed list'
+			// like [hello-dolly/hello.php] => 1369572703 so we do array_keys to make the array
+			// compatible with $active_plugins
+			$active_sitewide_plugins = (array) array_keys( get_site_option( 'active_sitewide_plugins', array() ) );
+			// merge arrays and remove doubles
+			$active_plugins = (array) array_unique( array_merge( $active_plugins, $active_sitewide_plugins ) );
+		}
+
+		return $active_plugins;
+	}
+	
+	public function deactivate_ubl_addon() {
+		$legacy_addon = $this->ubl_addon_detected();
+		if ( ! empty( $legacy_addon ) ) {
+			deactivate_plugins( $legacy_addon );
+			set_transient( 'wpo_wcpdf_ubl_addon_detected', 'yes', DAY_IN_SECONDS );
+		}
+	}
+	
+	public function ubl_addon_detected() {
+		$active_plugins = $this->get_active_plugins();
+		$ubl_addon      = '';
+		
+		foreach ( $active_plugins as $plugin ) {
+			if ( strpos( $plugin, 'ubl-woocommerce-pdf-invoices.php' ) !== false ) {
+				$ubl_addon = $plugin;
+				break;
+			}
+		}			
+		
+		return $ubl_addon;
+	}
+	
+	public function ubl_addon_active() {
+		if ( get_transient( 'wpo_wcpdf_ubl_addon_detected' ) ) {
+			ob_start();
+			?>
+			<div class="notice notice-warning">
+				<p><?php _e( 'The UBL addon was installed in your store but is no longer needed. We have disable it when you upgraded the PDF Invoices & Packing Slips for WooCommerce plugin. You are free to uninstall it.', 'woocommerce-pdf-invoices-packing-slips' ); ?></p>
```

Comment:
I would suggest changing this to: `While updating the ‘PDF Invoices & Packing Slips for WooCommerce’ plugin we’ve noticed our UBL add-on was active on your site. This functionality is now incorporated into the core plugin. We’ve deactivated the add-on for you, and you are free to uninstall it.`